docs/app-debug/debug-options.rst

```diff
@@ -0,0 +1,72 @@
+Open MPI Runtime Debugging Options
+==================================
+
+Open MPI has a series of MCA parameters for the MPI layer
+itself that are designed to help with debugging.
+These parameters :ref:`can be set <label-running-setting-mca-param-values>`
+in the usual ways.  MPI-level MCA parameters can be
+displayed by invoking the following command:
+
+.. code-block:: sh
+
+   # Use "--level 9" to see all the MCA parameters
+   # (the default is "--level 1"):
+   shell$ ompi_info --param mpi all --level 9
+
+Here is a summary of the debugging parameters for the MPI layer:
+
+* ``mpi_param_check``: If set to true (any positive value), and when
+  Open MPI is compiled with parameter checking enabled (the default),
+  the parameters to each MPI function can be passed through a series
+  of correctness checks.  Problems such as passing illegal values
+  (e.g., NULL or ``MPI_DATATYPE_NULL`` or other "bad" values) will be
+  discovered at run time and an MPI exception will be invoked (the
+  default of which is to print a short message and abort the entire
+  MPI job).  If set to false, these checks are disabled, slightly
+  increasing performance.
+
+* ``mpi_show_handle_leaks``: If set to true (any positive value),
+  Open MPI will display lists of any MPI handles that were not freed before
+  ``MPI_FINALIZE`` (e.g., communicators, datatypes, requests, etc.)
+
+* ``mpi_no_free_handles``: If set to true (any positive value), do not
+  actually free MPI objects when their corresponding MPI "free"
+  function is invoked (e.g., do not free communicators when
+  ``MPI_COMM_FREE`` is invoked).  This can be helpful in tracking down
+  applications that accidentally continue to use MPI handles after
+  they have been freed.
+
+* ``mpi_show_mca_params``: If set to true (any positive value), show a
+  list of all MCA parameters and their values during ``MPI_INIT``.
```

Comment:
```suggestion
  list of all MCA parameters and their values during :ref:`MPI_Init(3) <mpi_init>`.
```

Reply:
Oops -- I missed this one.  I'd rephrase to say something like "when MPI is initialized" (because it's not just MPI_INIT anymore).